plugins/modules/ndo_fex_device.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Shreyas Srish (@shrsr) <ssrish@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_fex_device
+short_description: Manage FEX Devices on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage FEX Devices on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is supported on ND v3.2 (NDO v4.4) and later.
+author:
+- Shreyas Srish (@shrsr)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a fabric resource policy template.
+    - This parameter or O(template_id) is required.
+    type: str
+  template_id:
+    description:
+    - The ID of the template.
+    - The template must be a fabric resource policy template.
+    - This parameter or O(template) is required.
+    type: str
+  name:
+    description:
+    - The name of the FEX Device.
+    type: str
+    aliases: [ fex_device ]
+  uuid:
+    description:
+    - The UUID of the FEX Device.
+    type: str
+    aliases: [ fex_device_uuid ]
+  description:
+    description:
+    - The description of the FEX Device.
+    type: str
+  fex_id:
+    description:
+    - The FEX ID.
+    - This parameter is required when O(state=present).
+    type: int
```

Comment:
dont know if it conflict with anything but should we add alias for id
```suggestion
    - This parameter is required when O(state=present).
    type: int
    aliases: [ id ]
```